src/utils/plugin-metrics.ts

```diff
@@ -0,0 +1,74 @@
+import { IllegalOperationError } from '../utils/utils'
+import { Hub, MetricMathOperations, PluginConfig, PluginConfigId } from './../types'
+import { createPosthog } from './../worker/vm/extensions/posthog'
+
+interface PluginMetrics {
+    pluginConfig: PluginConfig
+    metrics: Record<string, number>
+}
+
+interface UpdateMetricPayload {
+    pluginConfig: PluginConfig
+    metricOperation: MetricMathOperations
+    metricName: string
+    value: number
+}
+
+export class PluginMetricsManager {
+    metricsPerPluginConfig: Record<PluginConfigId, PluginMetrics>
+
+    constructor() {
+        this.metricsPerPluginConfig = {}
+    }
+
+    sendPluginMetrics(hub: Hub): void {
+        for (const pluginConfigMetrics of Object.values(this.metricsPerPluginConfig)) {
+            const config = pluginConfigMetrics.pluginConfig
+            const posthog = createPosthog(hub, config)
+            posthog.capture(`$plugin_metrics`, {
```

Comment:
For the other 'internal' metrics that Karl did we used double $$, then also put it on a separate organisation (as those were for self hosted only). Maybe in this case we should filter them out in event_definition somehow.

Reply:
we could just not create an event definition for it in the first place

Reply:
Updated it here and in the UI PR to use `$$plugin_metrics`. Also added a list of events we want to skip creating definitions for - will probably be used for more events in the future. Would have filtered out everything with `$$` but maybe that's too much.

Reply:
Yeah I wouldn't do that as someone might just created an event with that and then get very confused!

Reply:
green light on the changes or?